apps/src/sites/code.org/pages/public/teacher-dashboard/index.js

```diff
@@ -729,32 +740,49 @@ function main() {
       }
     );
 
-    $scope.genericError = function (result) {
-      $window.alert("An unexpected error occurred, please try again. If this keeps happening, try reloading the page.");
-    };
+    $scope.tab = 'progress';
+    $scope.page = {zoom: false};
+    $scope.react_progress = false;
 
+    // We want to make our sections service request whether using react or angular
+    // because our tabs haml (in nav.haml) depends on the section being loaded. It
+    // could probably be cleaned up to behave differently without too much difficulty,
+    // which would make this unnecessary.
     $scope.section = sectionsService.get({id: $routeParams.id});
+    // sections is used by our section dropdown, which at least initially will
+    // still be in angular
     $scope.sections = sectionsService.query();
-    const paginatedPromise = paginatedSectionProgressService.get($routeParams.id)
-      .then(result => {
-        $scope.progress = result;
-      })
-      .catch($scope.genericError);
-
-    $scope.tab = 'progress';
-    $scope.page = {zoom: false};
 
     // error handling
+    $scope.genericError = function (result) {
+      $window.alert("An unexpected error occurred, please try again. If this keeps happening, try reloading the page.");
+    };
     $scope.section.$promise.catch($scope.genericError);
     $scope.sections.$promise.catch($scope.genericError);
 
     // the ng-select in the nav compares by reference not by value, so we can't just set
     // selectedSection to section, we have to find it in sections.
-    $scope.sections.$promise.then(
-      function ( sections ){
-        $scope.selectedSection = $.grep(sections, function (section) { return (section.id == $routeParams.id);})[0];
-      }
-    );
+    $scope.sections.$promise.then(sections => {
+      $scope.selectedSection = sections.find(section => section.id.toString() === $routeParams.id);
+    });
+
+    if (experiments.isEnabled('sectionProgressRedesign')) {
+      $scope.react_progress = true;
+      $scope.$on('section-progress-rendered', () => {
+        $scope.section.$promise.then(script =>
+          renderSectionProgress(script, valid_scripts)
+        );
+      });
+      return;
+    }
+
+    // The below is not run if our experiment is not enabled
```

Comment:
Let's be more specific about the experiment here. There are a few going at a time in teacher dashboard.